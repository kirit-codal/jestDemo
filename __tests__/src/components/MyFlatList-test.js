import React from 'react';
import {FlatList, Text} from 'react-native';
import renderer from 'react-test-renderer';

const data = [
  {id: '1', title: 'Item 1'},
  {id: '2', title: 'Item 2'},
  {id: '3', title: 'Item 3'},
];

test('renders FlatList with correct data', () => {
  const component = renderer.create(
    <FlatList
      data={data}
      renderItem={({item}) => <Text>{item.title}</Text>}
      keyExtractor={item => item.id}
    />,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
