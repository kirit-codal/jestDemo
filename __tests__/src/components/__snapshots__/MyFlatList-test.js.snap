// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FlatList with correct data 1`] = `
<RCTScrollView
  data={
    [
      {
        "id": "1",
        "title": "Item 1",
      },
      {
        "id": "2",
        "title": "Item 2",
      },
      {
        "id": "3",
        "title": "Item 3",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item 1
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item 2
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item 3
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
